### PR TITLE
Log level for secondary workers.

### PR DESCRIPTION
**Pull request by tdomhan**
Allow specifying a different log level for secondary workers.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by fhieber — APPROVED** [review]
thanks!

**Comment by mjdenkowski** [comment]
Thanks for adding this functionality!

I recommend that the default log level for secondary workers is the same as for the primary worker.  This creates a lot of log lines, but it lets the user know that all processes are running as expected.  This is a useful default behavior for when users are first setting up Sockeye, Horovod, and MPI on one or more hosts.  After the user confirms everything is working, they can specify one of the options to reduce console logging.  This is the approach we use for the `--quiet-secondary-workers` option.

**Comment by fhieber** [comment]
Looks like `test_arguments` requires an update for the checks to pass.

File: sockeye/log.py [hunk 1]
```diff
@@ -128,6 +130,9 @@ def setup_main_logger(file_logging=True, console=True, path: Optional[str] = Non
     for _, handler_config in log_config['handlers'].items():  # type: ignore
         handler_config['level'] = level
 
+    if 'console' in log_config['handlers'] and console_level is not None:
```
Comment (fhieber) [inline sockeye/log.py:133]: you might need a `# type: ignore` somewhere here for mypy to pass

**Review by fhieber — CHANGES_REQUESTED** [review]

File: sockeye/log.py (hunk 1 above)
Comment (tdomhan) [reply sockeye/log.py]: Yeah, I did that but hadn't noticed that my push was rejected because I needed to pull the master merge :)

**Review by fhieber — APPROVED** [review]

**Pull request merged**